photutils/psf/photometry.py

```diff
@@ -235,22 +247,22 @@ def do_photometry(self, image, positions=None):
                           names=('x_0', 'y_0', 'flux_0'),
                           dtype=('f8', 'f8', 'f8'))
 
-            sources = self.find(residual_image)
+            sources = self.finder(residual_image)
 
             apertures = CircularAperture((sources['xcentroid'],
                                           sources['ycentroid']),
                                          r=self.aperture_radius)
 
             sources['aperture_flux'] = aperture_photometry(residual_image, apertures)['aperture_sum']
             n = 1
-            while(n <= self.niters and len(sources) > 0):
+            while(len(sources) > 0 and (self.niters is not None and n <= self.niters)):
```

Comment:
good point! it's actually supposed to be ``while(len(sources) > 0 and (self.niters is None or n <= self.niters))``, I think.  Will fix.